File: Java-version/LeetCode/src/com/jooc/GazeOffer/Solution_6.java
package com.jooc.GazeOffer;

import com.jooc.Solution;
import com.jooc.components.ListNode;

import java.util.Arrays;
import java.util.LinkedList;

public class Solution_6 implements Solution {
    @Override
    public void main() {
        Solution solution = new Solution();

        ListNode target = new ListNode(1);
        ListNode target2 = new ListNode(3);
        ListNode target3 = new ListNode(2);

        target.next = target2;
        target2.next = target3;

        System.out.println(Arrays.toString(solution.reversePrint(target)));
    }

    class Solution {
        public int[] reversePrint(ListNode head) {
            LinkedList<Integer> stack = new LinkedList<Integer>();

            while(head != null){
                stack.push(head.val);
                head = head.next;
            }

            int[] ans = new int[stack.size()];
            int index = 0;
            while(!stack.isEmpty()){
                ans[index++] = stack.pop();
            }

            return ans;
        }
    }
}

File: Swift-version/LeetCode/LeetCode/Solutions/GazeOffer/Solution_6.swift
//
//  Solution_6.swift
//  LeetCode
//
//  Created by 齐旭晨 on 2021/3/24.
//

import Foundation

class ListNode {
    public var val: Int
    public var next: ListNode?
    
    public init(_ val: Int) {
        self.val = val
        self.next = nil
    }
}

class Solution_6: Solution{
    func main() {
        
        let target = ListNode(1)
        target.next = ListNode(2)
        target.next?.next = ListNode(3)
        
        print(reversePrint(target))
    }
    
    func reversePrint(_ head: ListNode?) -> [Int] {
        var res:[Int] = []
        
        if  var node = head{
            res.append(node.val)
            while node.next != nil{
                node = node.next!
                res.append(node.val)
            }
        }
        
        return res.reversed()
    }
    
}
